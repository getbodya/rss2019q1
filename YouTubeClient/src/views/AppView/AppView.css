@keyframes stripes {
  0% {background-position: 0 0;}
  100% {background-position: 50px 0;}
}
* {
  margin: 0;
  font-family: Arial, Helvetica, sans-serif;
}
body {
  background-image: linear-gradient(#666 30% , #fff 85%);
  background-repeat: no-repeat;
}
.app {
  position: relative;
  width: 100%;
  height: 100vh;
}
.app__search-panel {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 10vh;
}
.search-panel__input {
  padding-left: 1%;
  width: 75%;
  height: 20px;
}
.search-panel__btn {
  cursor: pointer;
  width: 70px;
  color: #888;
  height: 26px;
}
.loading{
  animation: stripes .75s infinite linear;
  background: linear-gradient(45deg, rgba(30,255,188,1) 25%, rgba(255,255,255,1) 25%, rgba(255,255,255,1) 50%, rgba(30,255,188,1) 50%,      rgba(30,255,188,1) 75%, rgba(255,255,255,1) 75%, rgba(255,255,255,1));
  background-size: 10px 10px;
  color: #FF50E5;
}
.app__app-label {
  text-align: center;
  cursor: pointer;
}
.app__search-result {
  margin: 0 auto;
  width: 90%;
  height: 450px;
  cursor: grab;
  overflow-x: hidden;
}
.error-box {
  background: red;
  width: 300px;
  position: absolute;
  top:50%;
  left: calc(50% - 150px);
}
@media only screen and (max-width: 500px) {
  .search-panel__btn {
    width: 40px;
  }
}
