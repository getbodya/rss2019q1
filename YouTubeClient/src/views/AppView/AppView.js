import Events from '../../controllers/Events';


export default class AppView {
  constructor() {
    this.structure = {
      tag: 'div',
      className: 'app',
      child: [{
        tag: 'div',
        className: 'app__search-panel',
        child: [
          {
            tag: 'input',
            className: 'search-panel__input',
            placeholder: 'SEARCH',
          },
          {
            tag: 'button',
            className: 'search-panel__btn',
            content: '<i class="fas fa-search"></i>',
          },
        ],
      },
      {
        tag: 'h1',
        className: 'app__app-label',
        content: 'YOUTUBE CLIENT',
      },
      {
        tag: 'div',
        className: 'app__search-result',
        child: [
          {
            tag: 'div',
            className: 'cards',
          },
        ],
      },

      ],
    };
  }

  static createLayout(structure) {
    const createElement = param => {
      const { tag, className, content, placeholder, child } = param;
      const elem = document.createElement(tag);
      elem.setAttribute('class', className);
      if (content) {
        elem.innerHTML = content;
      }
      if (placeholder) {
        elem.setAttribute('placeholder', placeholder);
      }
      if (child) {
        child.forEach(item => {
          elem.appendChild(createElement(item));
        });
      }
      return elem;
    };
    return createElement(structure);
  }

  renderLayout() {
    const appLayout = AppView.createLayout(this.structure);
    document.body.appendChild(appLayout);
    Events.startSearch(document.querySelector('.search-panel__input'), 'keydown');
    Events.startSearch(document.querySelector('.search-panel__btn'), 'click');
    Events.imposeSwipe();
    Events.imposeTouchSwipe();
  }
}
