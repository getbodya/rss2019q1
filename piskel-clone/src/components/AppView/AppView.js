import ViewInstance from "../instances/ViewInstance";

export default class AppView extends ViewInstance{
  constructor(){
    super();
  }
}
