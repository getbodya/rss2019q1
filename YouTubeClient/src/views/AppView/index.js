import AppView from './AppView';

import './AppView.css';

export default AppView;
