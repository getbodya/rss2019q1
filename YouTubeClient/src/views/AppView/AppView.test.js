import AppView from './AppView';

const structure = {
  tag: 'div',
  className: 'app',
  child: [{
    tag: 'div',
    className: 'app__search-panel',
    child: [
      {
        tag: 'input',
        className: 'search-panel__input',
        placeholder: 'SEARCH',
      },
      {
        tag: 'button',
        className: 'search-panel__btn',
        content: '<i class="fas fa-search"></i>',
      },
    ],
  },
  {
    tag: 'h1',
    className: 'app__app-label',
    content: 'YOUTUBE CLIENT',
  },
  {
    tag: 'div',
    className: 'app__search-result',
    child: [
      {
        tag: 'div',
        className: 'cards',
      },
    ],
  },

  ],
};
describe('AppView.createLayout', () => {
  it('should return HTML element', () => {
    const res = AppView.createLayout(structure);
    expect(res).toBeInstanceOf(HTMLElement);
  });
});
describe('AppView.prototype.renderLayout', () => {
  it('should be renderLayout correctly', () => {
    const state = {
      structure,
    };
    AppView.prototype.renderLayout.call(state);
    expect(document.body.innerHTML).toMatchSnapshot();
  });
});
