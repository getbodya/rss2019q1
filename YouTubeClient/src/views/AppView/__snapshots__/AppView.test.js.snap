// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppView.prototype.renderLayout should be renderLayout correctly 1`] = `"<div class=\\"app\\"><div class=\\"app__search-panel\\"><input class=\\"search-panel__input\\" placeholder=\\"SEARCH\\"><button class=\\"search-panel__btn\\"><i class=\\"fas fa-search\\"></i></button></div><h1 class=\\"app__app-label\\">YOUTUBE CLIENT</h1><div class=\\"app__search-result\\"><div class=\\"cards\\"></div></div></div>"`;
